controller/nc_news-controllers.js

```diff
@@ -26,8 +26,25 @@ exports.sendAllArticles = (req, res, next) => {
 };
 
 exports.sendAllCommentsByID = (req, res, next) => {
-    const { article_id } = req.params
-    return selectAllCommentsByID(article_id).then((comments)=>{
+    const { article_id } = req.params;
+    return selectArticleByID(article_id).then(() =>{
+        return selectAllCommentsByID(article_id).then(( comments)=>{
         res.status(200).send({comments: comments})
-    }).catch(next)
+    })
+    }).catch((err) => {
+        next(err)
+    })
+}
+
+exports.postCommentsByID = (req, res, next) =>{
+    const { article_id } = req.params;
+    const comment = req.body;
+    return addCommentByID(comment, article_id).then((comment) => {
+        res.status(200).send({comment: comment});
+    }).catch((err) => {
+        if(err.code === "23503") {
```

Comment:
It would be better if this if statement weren't here, it should instead be handled in your error handling middleware function which is dealing with PSQL errors (line 35 in `app.js`)
If we funnel all errors to the error handling middleware by just invoking errors with `next`, it saves us having to repeat ourselves when the same error happens across multiple endpoints.